rdmo/accounts/tests/helpers.py

```diff
@@ -0,0 +1,140 @@
+import sys
+from importlib import import_module, reload
+from typing import Optional
+
+import pytest
+
+from django.core.management import call_command
+from django.db import connection
+from django.urls import clear_url_caches, get_resolver, reverse
+
+
+def reload_urlconf(urlconf=None, settings=None):
+    clear_url_caches()
+
+    if urlconf is None and settings is None:
+        from django.conf import settings
+        urlconf = settings.ROOT_URLCONF
+    elif urlconf is None and settings is not None:
+        # take the settings during pytest run
+        urlconf = settings.ROOT_URLCONF
+
+    if urlconf in sys.modules:
+        reload(sys.modules[urlconf])
+    else:
+        import_module(urlconf)
+
+
+def reload_urls(app_name: Optional[str] = None,
+                settings=None,
+                other_apps: list[str] | None = None ) -> None:
+    # reload the urlconf of the app
+    if app_name is not None:
+        reload_urlconf(urlconf=f'rdmo.{app_name}.urls', settings=settings)
+
+    # reload the core urlconf
+    reload_urlconf(urlconf='rdmo.core.urls', settings=settings)
+
+    # reload the testcase settings urlconf
+    reload_urlconf(settings=settings)
+
+    if other_apps:
+        for _app in other_apps:
+            reload_urlconf(urlconf=f"{_app}.urls", settings=settings)
+
+    get_resolver()._populate()
+
+
+@pytest.fixture(autouse=False)
+def enable_terms_of_use(settings): # noqa: PT004
+    settings.ACCOUNT_TERMS_OF_USE = True
+    settings.MIDDLEWARE += [
+        settings.ACCOUNT_TERMS_OF_USE_MIDDLEWARE
+    ]
+    reload_urls('accounts')  # Reloads URLs to apply new settings
+
+    yield
+
+    # revert settings to initial state
+    settings.ACCOUNT_TERMS_OF_USE = False
+    settings.MIDDLEWARE.remove(settings.ACCOUNT_TERMS_OF_USE_MIDDLEWARE)
+    # 🔹 Reload URLs to reflect the changes
+    reload_urls("accounts")
+
+
+@pytest.fixture(autouse=False)
+def enable_socialaccount(db, client, settings, django_db_setup, django_db_blocker, django_db_use_migrations): # noqa: PT004
```

Comment:
this got a bit messy but I hope now all the tests pass

Reply:
there were some issue with py38 incompatible style, but we are not using typing anyways..

Reply:
a fixture that adds an apps and needs migrations (the `allauth.socialaccount`) was not gonna work out.
So we've decided to add it to the initial test config settings `INSTALLED_APPS`, since an added table should not bother the other tests.